HIP/hip-751.md

```diff
@@ -0,0 +1,337 @@
+
+Claimable Balance
+
+Stellar's claimable balance feature resembles token locking by reserving payments for recipients, preventing reclamation by senders. However, it has limitations compared to the proposed feature in this HIP. Creating and managing separate claimable balance records adds complexity. The release of claimable balances is limited to specific conditions, and visibility of locked tokens in user wallets is not inherently supported, potentially causing confusion about token holdings.
+
+These approaches offer similar functionality to the proposed token locking mechanism, albeit with varying visibility, accessibility, and complexity levels. The Hedera Token Service's introduction of token locking will provide a standardized, easy-to-use, native mechanism for token holders to lock and unlock their tokens.
+
+## User Stories
+
+**_Scenario_** - _Token Issuer_: Token issuance with transfer lock enabled 
+
+AS A _Token Issuer_,
+
+I WANT to be able to enable the transfer lock to a portion of my fungible tokens as needed 
+
+SO THAT I can control the transferability of tokens and avoid token holders trading and transferring during the lock up period or otherwise as needed.
+
+AS A _Token Issuer_, 
+
+I WANT to be able to lock a portion of my issued fungible tokens for maintenance or or other reasons as needed,
+
+SO THAT I can ensure the integrity and security of the tokens.
+
+AS A _Token Issuer_,
+
+I WANT to be able to transfer unlock a portion of fungible token supply according to their respective unlocking conditions, 
+
+SO THAT I can maintain control over the transferring of the token supply.
+
+AS A _Token Issuer_,
+
+I WANT to be able to set a transfer lock key and to activate the transfer lock function to a portion of my fungible token supply, 
+
+SO THAT my tokens are customized to meet my specific needs, regulatory or otherwise.
+
+
+**_Scenario_** - _Token Issuer_: Enabling customizable execution time for transfer lock on a portion of token supply 
+
+AS A a _Token Issuer_,
+
+I WANT to be able to implement a duration-based token transfer lock for a portion of the total supply, 
+
+SO THAT so that I can ensure token holders can not trade the supply unless the lock time has expired. 
+
+AS A _Token Issuer_,
+
+I WANT to have the option to enable the transfer lock function for a portion of my fungible token supply with a predetermined execution time,
+
+SO THAT I can schedule the unlocking of tokens in advance without the need for immediate manual execution.
+
+**_Scenario_** -  _Token Issuer_: Executing native events on the locked portion of a token supply
+
+AS A _Token Issuer_,
+
+I WANT to be able to execute events (metadata update, mint, burn, transfers and more) on my token even when the lock functionality is enabled, 
```

Comment:
For clarification, it sounds like a token _issuer_ can still transfer locked tokens, but token _holders_ can't exchange these locked tokens. Did I get that right?

Reply:
@mhess-swl - Yes, that is correct. The owner of the LockKey can execute events on the tokens even while they are in a locked state, including transfers.  The lock primitive, as we envision it, targets a specific account, the account that should be the token holder will not be able to successfully execute the transfer till the lock has expired.

Reply:
Understood, thank you!